### DISCUSSION TITLE
Adding `/var/labels` folder to default configuration

### DISCUSSION THREAD

**Discussion by Moongazer** [sourcebroker/deployer-extended-typo3#14] [Ideas]
The `/var/labels` folder is used to store the label files downloaded by [Manage Language Packs](https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/ApiOverview/Internationalization/ManagingTranslations.html).

Currently after each deployment, all language packs have to be manually downloaded again, because this folder is not set in `shared_dirs`. One could add this to their custom Deployer config, but IMO it make sense to add this to the default `shared_dirs` config similar to `/var/log` and `/var/transient`:
```
set('shared_dirs', function () {
    return [
        // ...
        !empty(get('web_path')) ? 'var/labels' : 'typo3temp/var/labels',
    ];
});
```
Alternatively, the deployment script should run `/path/to/typo3/bin/typo3 language:update` after changing the deployment-dir, to trigger the full download of all the files, if the folder should stay at its current place (`/current/var/labels`).

**Comment by kszymukowicz** [comment]
This is arguable if:
1. `/var/labels` should be in shared or not
2. if in shared should it be updated on each deploy

**In core documentation the `/var/labels` is not shared:**
https://docs.typo3.org/m/typo3/tutorial-getting-started/master/en-us/Installation/DeployTYPO3.html#deployment-automation

**TYPO3 Core Distribution also have `/var/labels` in git:**
https://github.com/TYPO3/TYPO3.CMS.BaseDistribution/blob/master/.gitignore

Updating lables on each deploy is IMO crazy idea. Labels used in frontend is something that your customer accepted during quality checks. Updating labels on each deploy means you expecting that each label version will be better than already existing, that there will be no bugs etc. So if we trust so much in other work then maybe lets update Bootstrap, lets update all TYPO3 extensions, TYPO3 core on each deploy without checking if all is working. (to be honest there is one TYPO3 magician called *jpmschuler* who updating everything above on each night, but then you need to have lot of frontend/functional testing to be sure everything is working as before)

IMO acceptable solution is to share it and update manually from backend level (and check if everything is working as expected after update) but then you "add" manual work on deployment, which means someone must remember that with deploy X (for example with new extension) there is need to run also language update on TYPO3 backend.

All this problems are gone when `/var/labels/` are in git.

My opinion is that `/var/labels/` should be in git and updated only when needed, for example when you updated extension or added new extension so for now there will be no changes in `shared_dirs`.
